src/php/Command/ReplCommand.php

```diff
@@ -21,102 +20,123 @@ final class ReplCommand
 
     private const ENABLE_BRACKETED_PASTE = "\e[?2004h";
     private const DISABLE_BRACKETED_PASTE = "\e[?2004l";
-    private const PROMPT = '>>> ';
+    private const INITIAL_PROMPT = '>>> ';
+    private const OPEN_PROMPT = '... ';
+    private const EXIT_REPL = 'exit';
 
     private ReplCommandIoInterface $io;
     private EvalCompilerInterface $compiler;
     private ExceptionPrinterInterface $exceptionPrinter;
     private ColorStyleInterface $style;
+    private PrinterInterface $printer;
+
+    /** @var string[] */
+    private array $inputBuffer = [];
 
     public function __construct(
         ReplCommandIoInterface $io,
         EvalCompilerInterface $compiler,
         ExceptionPrinterInterface $exceptionPrinter,
-        ColorStyleInterface $style
+        ColorStyleInterface $style,
+        PrinterInterface $printer
     ) {
         $this->io = $io;
         $this->compiler = $compiler;
         $this->exceptionPrinter = $exceptionPrinter;
         $this->style = $style;
+        $this->printer = $printer;
     }
 
     public function run(): void
     {
         $this->io->readHistory();
-        $this->io->output($this->style->yellow("Welcome to the Phel Repl\n"));
-        $this->io->output('Type "exit" or press Ctrl-D to exit.' . "\n");
+        $this->io->output($this->style->yellow('Welcome to the Phel Repl' . PHP_EOL));
+        $this->io->output('Type "exit" or press Ctrl-D to exit.' . PHP_EOL);
 
         $this->loopReadLineAndAnalyze();
     }
 
-    /**
-     * @throws ExitException
-     */
     private function loopReadLineAndAnalyze(): void
     {
         while (true) {
             try {
-                if ($this->io->isBracketedPasteSupported()) {
-                    $this->io->output(self::ENABLE_BRACKETED_PASTE);
-                }
-                $input = $this->io->readline(self::PROMPT);
-                if ($this->io->isBracketedPasteSupported()) {
-                    $this->io->output(self::DISABLE_BRACKETED_PASTE);
-                }
-
-                try {
-                    $this->checkInputAndAnalyze($input);
-                } catch (ExitException $e) {
-                    break;
-                }
+                $this->addLineFromPromptToBuffer();
+                $this->checkExitInputBuffer();
+                $this->analyzeInputBuffer();
+            } catch (ExitException $e) {
+                break;
             } catch (Throwable $e) {
-                $this->io->output($e->getTraceAsString() . PHP_EOL);
+                $this->inputBuffer = [];
+                $this->io->output($this->style->red($e->getMessage() . PHP_EOL));
```

Comment:
Why are we back to `->getMessage()` only. Shouldn't it be `->getTraceAsString()`?

Reply:
Ups, right, I think I missed that. I will put the trace back